### PR TITLE
CRS-5558 Add rails 7 support in ssm_config

### PR DESCRIPTION
**Pull request by Ifraah**
Closes https://snapsheettech.atlassian.net/browse/CRS-5558

Support rails 7.

Test locally using instructions in - https://github.com/bodyshopbidsdotcom/hermes/pull/173

### CODE REVIEW

File: ssm_config.gemspec [hunk 1]
```diff
@@ -30,7 +30,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake', '~> 10.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rspec-rails'
-  spec.add_dependency 'rails', '>= 3', '< 7'
+  spec.add_dependency 'rails', '>= 3', '< 7.1'
```
Comment (lluchez) [inline ssm_config.gemspec:33]: You can probably do `< 8` or even drop the restriction

**Review by lluchez — APPROVED** [review]
LG 👍

File: lib/ssm_config.rb [hunk 2]
```diff
@@ -3,11 +3,10 @@
 require 'ssm_config/ssm_storage/empty.rb'
 require 'ssm_config/errors.rb'
 require 'ssm_config/migration_helper.rb'
-require 'active_support/core_ext/hash/indifferent_access'
-require 'active_support/time'
+require 'active_support/all'
```
[the remark this answers is not in the document]
Comment (holmes2) [reply lib/ssm_config.rb:6]: can you try with this `require "active_support/hash_with_indifferent_access"` and check if this does not fail ?

My guess is that active support 7 has this pathing for `hash_with_indifferent_access` .. but the earlier ones might be different.

If it does not fail, then i think keeping it `active_support/all` might be better to support all versions of rails

[the remark this answers is not in the document]
Comment (holmes2) [reply lib/ssm_config.rb:6]: Checked with TX for `active_support/all` ... with Rails 6 it is working as expected ✅ 
image

File: lib/ssm_config.rb [hunk 3]
```diff
@@ -3,11 +3,10 @@
...
 
 module SsmConfig
-  VERSION = '1.3.3'.freeze
+  VERSION = '1.3.4'.freeze
   REFRESH_TIME = (30.minutes).freeze
```
Comment (holmes2) [inline lib/ssm_config.rb:10]: Do u folks want to change refresh time to an ENV variable ? This might be helpful against an ongoing issue with ssm_config

Comment (lluchez) [reply lib/ssm_config.rb:10]: @holmes2 thanks for the suggestion, we don't use the new features that you guys added so far on `ssm_config`. We might eventually, but we already have [SystemVariables](https://github.com/bodyshopbidsdotcom/snapsheet/blob/remove_gem_text/app/models/system_variable.rb#L13) which are solving similar issues.

Also, we should keep this PR only for Rails 7.

Thanks

File: lib/ssm_config.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ifraah) [reply lib/ssm_config.rb:6]: Verified that `require "active_support/hash_with_indifferent_access"` Does not fail for rails 7

**Review by holmes2 — APPROVED** [review]

**Pull request merged**